core/io/primitives.h

```diff
@@ -30,8 +31,14 @@ namespace Gadgetron::Core::IO {
     void read(std::istream &stream, std::set<T> &vec);
 
     template<class T>
-    void read(std::istream &stream, hoNDArray<T> &array);
+    void read(std::istream &stream, Gadgetron::hoNDArray<T> &array);
 
+    template<class T, unsigned int D>
+    void read(std::istream &stream, Gadgetron::hoNDImage<T, D> &array);
+
+    template<class T, unsigned int D>
+    void read(std::istream &stream, Gadgetron::hoNDArray< Gadgetron::hoNDImage<T, D> > &array);
```

Comment:
Same as above, I will not repeat this comment throughout, but through all the read and write function, you use variable name `array`, when it should be `image`

Reply:
fixed